api/types/provisioning.go

```diff
@@ -394,6 +395,11 @@ func (p *ProvisionTokenV2) GetJoinMethod() JoinMethod {
 	return p.Spec.JoinMethod
 }
 
+// IsStatic returns true if the token is statically configured
+func (p *ProvisionTokenV2) IsStatic() bool {
+	return p.Origin() == OriginConfigFile
+}
+
```

Comment:
Ok, I added the `IsStatic` method but I am not relying on the time anymore (it seemed fickle to me from the start). When we fetch static tokens, we create a new `ProvisionToken` from the available static tokens. I've just set their origin to `OriginConfiguration` instead of being nil and now `IsStatic` just checks if `Origin() == OriginConfiguration`. It seems to be more correct and resilient this way. Let me know what you think